src/lib/default.nix

```diff
@@ -1,10 +1,12 @@
 {
   lib,
-  config ? (import ../utils/config.nix).loadConfig {},
+  config ? {},
   ...
-}: let
+} @ args: let
   l = lib // builtins;
 
+  config = (import ../utils/config.nix).loadConfig args.config or {};
```

Comment:
I moved this here because this file receives an "unloaded" config from `src/default.nix`. Otherwise some code would give errors related to a config that must be set is not set.